tcp_server.js

```diff
@@ -0,0 +1,22 @@
+var net = require('net');
+var fs = require('fs');
+var chai = require('chai'), chaiHTTP = require('chai-http');
+
+chai.use(chaiHTTP);
+
+var server = net.createServer(function(socket){
+  socket.on('data', function(data){
+    console.log(data.toString());
+  });
+
+  socket.on('end', function(){
+    console.log('socket closed');
+    server.close();
```

Comment:
The 'end' event will be emitted when the socket is closed, so closing it in the 'end' event is redundant.